content/blog/scheduling-in-react/index.md

```diff
@@ -47,29 +47,33 @@ With the information above, we can formulate two problems that we have to solve
 
 _⚠️ Warning: The following APIs are not yet stable and will change. I will do my best to keep this post updated._
 
-To implement a properly scheduled user interface with React, we have to look into two new React features:
+To implement a properly scheduled user interface with React, we have to look into two upcoming React features:
 
 - **Concurrent React (also known as Time Slicing).** With the help of the new [Fiber architecture](https://www.youtube.com/watch?v=ZCuYPiUIONs) rewrite that was released with React 16, React can now pause during rendering and yield[^2] to the main thread.
 
+  We will hear more about Concurrent React in the future but for now it is important to understand that when this mode is enabled, React will split the synchronous rendering of our React components into pieces that are run over multiple frames.
+
   ➡️ With this feature, we’re able to split long-running rendering tasks into small chunks.
 
-- **Scheduler.** This is a general purpose cooperative main thread scheduler developed by the React Core team that makes it possible to register callbacks with different priority levels in the browser.
+- **Scheduler.** The general general purpose cooperative main thread scheduler is developed by the React Core team and makes it possible to register callbacks with different priority levels in the browser.
 
   At the time of writing this article, the priority levels are:
 
   - `Immediate` for tasks that need to run synchronously.
   - `UserBlocking` (250ms timeout) for tasks that should run as the result of a user interaction (e.g. a button click).
-  - `Normal` (5s timeout) for rendering non-interactive parts of the screen.
+  - `Normal` (5s timeout) for updates that don’t have to feel instantaneous.
   - `Low` (10s timeout) for tasks that can be deferred but must still complete eventually (e.g. an analytics notification).
   - `Idle` (no timeout) for tasks that do not have to run at all (e.g. hidden offscreen content).
 
   The timeouts for each priority level are necessary to make sure that lower priority work still runs even if we have so much higher priority work to do that the higher priority work could run continuously. In scheduling algorithms, this problem is referred to as [starvation](<https://en.wikipedia.org/wiki/Starvation_(computer_science)>). The timeouts give us the guarantee that every scheduled task will eventually run. For example, we won’t miss a single analytics notification, even if we have ongoing animations in our app.
 
+  Under the hood, the Scheduler will store all registered callbacks in a list ordered by the expiration time (which is the time at which the callback was registered plus the timeout of the priority level). Then, the Scheduler will itself register a callback that is run after the next frame is drawn by the browser.[^3] Within this callback, the Scheduler will execute as many of the registered callbacks until it’s time to render the next frame.
```

Comment:
I think I see this more clear now. Maybe just complete with "...the Scheduler will execute as many of the registered callbacks **as possible** until it’s time..."